tm-parsers/js/js.tm

```diff
@@ -1108,8 +1125,8 @@ YieldExpression<In, Await> -> Yield :
 AsyncMethod<Yield, Await> -> AsyncMethod :
     'async' .afterAsync .noLineBreak PropertyName ('?'|'!')? UniqueFormalParameters<~Yield, +Await> AsyncFunctionBody ;
 
-AsyncFunctionDeclaration -> AsyncFunc :
-    'async' .afterAsync .noLineBreak 'function' BindingIdentifier? FormalParameters<~Yield, +Await> AsyncFunctionBody ;
```

Comment:
+Await in FormalParameters is correct. See https://tc39.es/ecma262/#prod-AsyncFunctionDeclaration

The rule for BindingIdentifier was simplified to always allow async/await - it does not change anything for parsing. Therefore, we can remove Await from AsyncFunctionDeclaration, and so from HoistableDeclaration too.